src/device/router_device_cache.erl

```diff
@@ -52,13 +59,24 @@ get(DeviceID) ->
 
 -spec get_by_devaddr(binary()) -> [router_device:device()].
 get_by_devaddr(DevAddr) ->
-    MS = ets:fun2ms(fun({_, D}) when D#device_v6.devaddr == DevAddr -> D end),
-    ets:select(?ETS, MS).
+    case ets:lookup(?DEVADDR_ETS, DevAddr) of
+        [] -> [];
+        List -> [Device || {_DevAddr, Device} <- List]
+    end.
 
 -spec save(router_device:device()) -> {ok, router_device:device()}.
 save(Device) ->
     DeviceID = router_device:id(Device),
     true = ets:insert(?ETS, {DeviceID, Device}),
+    % MS = ets:fun2ms(fun({_, D}) when D#device_v7.id == DeviceID -> true end),
+    MS = [{{'_', '$1'}, [{'==', {element, 2, '$1'}, {const, DeviceID}}], [true]}],
```

Comment:
What are the semantics of `MS` in this context?  Nothing was apparent from sibling functions.  Please add a code comment.